src/dom.ts

```diff
@@ -20,6 +20,50 @@ for (const param in tagWrap) {
 	tw.post = '</' + tw.reverse().join('></') + '>';
 }
 
+function validateClass(token: string): void {
+	if (!token || token === '') {
+		throw new Error('An invalid or illegal string was specified');
+	}
+	if (/\s/.test(token)) {
+		throw new Error('String contains an invalid character');
+	}
+}
+
+/**
+ * Adds one or more CSS class names to an Element without duplication
+ *
+ * @param element The Element to which to add CSS classes
+ * @param classes An array of string CSS classes to add to the Element
+ *
+ * @example
+ * dom.addClass(document.body, 'loaded');
+ *
+ * @example
+ * dom.addClass(document.body, 'loaded', 'ready');
+ */
+export function addClass(element: Element, ...classes: string[]): void {
+	let targetElement = <any> element;
+	if (!targetElement) {
+		return;
+	}
+	if (targetElement.classList) {
+		let classList: any = targetElement.classList;
+		classList.add.apply(classList, classes);
+	}
+	else {
+		let newClasses: string[] = new Array();
+		for (let className in classes) {
+			validateClass(className);
+			if (!this.contains(targetElement, className)) {
+				newClasses.push(className);
+			}
+		}
+		if (newClasses.length > 0) {
+			targetElement.className += (' ' + newClasses.join(' '));
```

Comment:
I'm nit-picking, but the initial space only needs to be added if `targetElement.className` is currently not empty.